### PR TITLE
Change Syncer API to return an Update object with additional metadata.

### PR DESCRIPTION
**Pull request by fasaxc**
Felix uses the new metadata to count keys as they churn.

### CODE REVIEW

File: lib/backend/model/keys.go [hunk 1]
```diff
@@ -87,6 +87,21 @@ type KVPair struct {
 	TTL      time.Duration // For writes, if non-zero, key has a TTL.
 }
 
+// Update from the Syncer.  A KV pair plus extra metadata.
+type Update struct {
```
Comment (robbrockbank) [inline lib/backend/model/keys.go]: wdyt?

**Comment by robbrockbank** [comment]
One comment/question/suggestion in line.  But nothing I feel super strongly about, so lgtm if you don't want to change it.

**Comment by fasaxc** [comment]
Done, back to you @robbrockbank

**Comment by robbrockbank** [comment]
lgtm

**Pull request merged**